patches/CreatePBP.cmake

```diff
@@ -12,73 +13,126 @@ cmake_minimum_required(VERSION 3.10)
+    if (NOT DEFINED ARG_${arg})
+      set(ARG_${arg} "NULL")
     endif()
+  endforeach()
+
+  if(NOT ${ARG_BUILD_PRX})
+    set(ARG_ENC_PRX FALSE BOOL)
+  endif()
 
+  if(${ARG_BUILD_PRX})
+    target_link_options(${ARG_TARGET}
+      PUBLIC -specs=${PSPSDK}/lib/prxspecs
+      PUBLIC -Wl,-q,-T${PSPSDK}/lib/linkfile.prx
+      PUBLIC ${PSPSDK}/lib/prxexports.o)
+  endif()
+
+  if("${CMAKE_BUILD_TYPE}" STREQUAL "Release" AND NOT ${ARG_BUILD_PRX})
+    add_custom_command(
+      TARGET ${ARG_TARGET}
+      POST_BUILD COMMAND
+      "${STRIP}" "$<TARGET_FILE:${ARG_TARGET}>"
+      COMMENT "Stripping binary"
+      )
+  elseif(${ARG_BUILD_PRX})
     add_custom_command(
-            TARGET ${ARG_TARGET} POST_BUILD
-            COMMAND ${CMAKE_COMMAND} -E make_directory
-            $<TARGET_FILE_DIR:${ARG_TARGET}>/psp_artifact
-            COMMENT "Creating psp_artifact directory."
+      TARGET ${ARG_TARGET}
+      POST_BUILD COMMAND
+      ${CMAKE_COMMAND} -E cmake_echo_color --cyan "Not stripping binary because building PRX."
+      )
+  else()
+    add_custom_command(
+      TARGET ${ARG_TARGET}
+      POST_BUILD COMMAND
+      ${CMAKE_COMMAND} -E cmake_echo_color --cyan "Not stripping binary, build type is ${CMAKE_BUILD_TYPE}."
+      )
+  endif()
+
+  add_custom_command(
+    TARGET ${ARG_TARGET} POST_BUILD
+    COMMAND ${CMAKE_COMMAND} -E make_directory
+    $<TARGET_FILE_DIR:${ARG_TARGET}>/psp_artifact
+    COMMENT "Creating psp_artifact directory."
     )
 
-    add_custom_command(
-            TARGET ${ARG_TARGET} POST_BUILD
-            COMMAND ${CMAKE_COMMAND} -E copy_if_different
-            $<TARGET_FILE:${ARG_TARGET}>
-            $<TARGET_FILE_DIR:${ARG_TARGET}>/psp_artifact
-            COMMENT "Copying ELF to psp_arfitact directory."
+  add_custom_command(
+    TARGET ${ARG_TARGET} POST_BUILD
+    COMMAND ${CMAKE_COMMAND} -E copy_if_different
+    $<TARGET_FILE:${ARG_TARGET}>
+    "$<TARGET_FILE_DIR:${ARG_TARGET}>/psp_artifact/${ARG_TARGET}.elf"
+    COMMENT "Copying ELF to psp_arfitact directory."
     )
 
-    add_custom_command(
-            TARGET ${ARG_TARGET}
-            POST_BUILD COMMAND
-            "${FIXUP}" "$<TARGET_FILE_DIR:${ARG_TARGET}>/psp_artifact/${ARG_TARGET}"
-            COMMENT "Calling psp-fixup-imports"
+  add_custom_command(
+    TARGET ${ARG_TARGET}
+    POST_BUILD COMMAND
+    "${FIXUP}" "$<TARGET_FILE_DIR:${ARG_TARGET}>/psp_artifact/${ARG_TARGET}.elf"
+    COMMENT "Calling psp-fixup-imports"
     )
 
+  if (${ARG_BUILD_PRX})
     add_custom_command(
-            TARGET ${ARG_TARGET}
-            POST_BUILD COMMAND
-            "${MKSFOEX}" "-d" "MEMSIZE=1" "${ARG_TITLE}" "PARAM.SFO"
-            COMMENT "Calling mksfoex"
-    )
+      TARGET ${ARG_TARGET}
+      POST_BUILD COMMAND
+      "${PRXGEN}" "$<TARGET_FILE_DIR:${ARG_TARGET}>/psp_artifact/${ARG_TARGET}.elf" "$<TARGET_FILE_DIR:${ARG_TARGET}>/psp_artifact/${ARG_TARGET}.prx"
+      COMMENT "Calling prxgen"
+      )
 
+    if(${ARG_ENC_PRX})
```

Comment:
Please replace with `if(${ARG_BUILD_PRX} AND ${ARG_ENC_PRX})` and remove lines 28-30

Reply:
Why? My intention was to have 2 separate flags - one for building PRX and one for encrypting it.

Reply:
It's just shorter to have it in one condition

Reply:
The point was that someone might not want to encrypt their prx

Reply:
Also if you dont want to encrypt prx that `AND` condition will fail and you get no PRX, just sad ELF...